From 8cda7249e84c5fd746975cf4c59bd54a7cdb75c2 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Tue, 8 May 2018 09:36:47 +0800
Subject: [PATCH 3736/5242] MLK-18222: CI_PI: fix CI_PI lost half of frame
 issue

commit  6ff717106cbd26b69cb6f849da6ec8ec95a19de9 from
https://source.codeaurora.org/external/imx/linux-imx.git

As IC team recommended, it is better to set vsync pulse
width as 2 lines pixels, otherwise ISI will lost half
of frames.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Reviewed-by: Sandor.Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-parallel-csi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx8/mxc-parallel-csi.c b/drivers/media/platform/imx8/mxc-parallel-csi.c
index f06a6b3..c821717 100644
--- a/drivers/media/platform/imx8/mxc-parallel-csi.c
+++ b/drivers/media/platform/imx8/mxc-parallel-csi.c
@@ -193,7 +193,7 @@ static void mxc_pcsi_config_ctrl_reg1(struct mxc_parallel_csi_dev *pcsidev)
 
 	/* Config Pixel Width */
 	val = (CSI_CTRL_REG1_PIXEL_WIDTH(pcsidev->format.width - 1) |
-		CSI_CTRL_REG1_VSYNC_PULSE(3));
+		CSI_CTRL_REG1_VSYNC_PULSE(pcsidev->format.width << 1));
 	writel(val, pcsidev->csr_regs + CSI_CTRL_REG1);
 
 }
-- 
1.7.9.5

